textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/LuceneDataReader.java

```diff
@@ -44,14 +40,11 @@
     private IndexSearcher indexSearcher;
     private ScoreDoc[] scoreDocs;
     private IndexReader indexReader;
-    private String query;
-    private String defaultField;
+    private Query query;
     
-    public LuceneDataReader(IDataStore dataStore, 
-            String query, String defaultField) {
+    public LuceneDataReader(IDataStore dataStore, Query query) {

```

Comment:
Another related question: why does Lucene make sure the analyzer in a reader is consistent with the analyzer for the index?  What if they are not?